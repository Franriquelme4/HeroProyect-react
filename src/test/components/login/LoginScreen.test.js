import { LoginScreen } from "../../../components/login/LoginScreen";
import React from 'react';
import { mount } from "enzyme";
import { AuthContext } from "../../../auth/authContext";
import { MemoryRouter, Route, Routes } from "react-router-dom";
import { types } from "../../../types/types";
const mockNavigate = jest.fn();
jest.mock('react-router-dom',()=>({
    ...jest.requireActual('react-router-dom'),
    useNavigate:()=>mockNavigate
}))
describe('Pruebas con <Login Screen', () => { 
    const contextValue = {
        dispatch:jest.fn(),
        user:{
                logged:false
            }
        }

//const contextValue = 
    const wrapper =  mount(
    <AuthContext.Provider value={contextValue}>
        <MemoryRouter initialEntries={['/login']}>
            <Routes >
                <Route path='/login' element={<LoginScreen/>}/>
            </Routes>
        </MemoryRouter>

    </AuthContext.Provider>
    );
    test('Debe de mostrarse correctamente', () => { 
        expect(wrapper).toMatchSnapshot();


     })
     test('Debe de realizar el dispach y la navegacion', () => { 
         const handleClick =  wrapper.find('button').prop('onClick');
         handleClick();
         expect(contextValue.dispatch).toHaveBeenCalledWith({
             type:types.login,
             payload:{name:'Francisco'}
         });
         expect(mockNavigate).toHaveBeenCalledWith('/',{replace:true});
         localStorage.setItem('lastPath','/dc');
         handleClick();
         expect(mockNavigate).toHaveBeenCalledWith('/',{replace:true});
      })
 })
