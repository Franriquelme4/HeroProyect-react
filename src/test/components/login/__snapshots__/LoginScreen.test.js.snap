// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con <Login Screen Debe de mostrarse correctamente 1`] = `
<div
  className="container mt-5"
>
  <h1>
    LoginScrean
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
